### PR TITLE
Brian/order settlement

### PR DESCRIPTION
**Pull request by bweick**

### CODE REVIEW

File: contracts/core/extensions/CoreIssuanceOrder.sol [hunk 1]
```diff
@@ -272,4 +275,57 @@ contract CoreIssuanceOrder is
...
+            requiredBalances.push(tokenBalance.add(requiredAddition));
+        }
+
+        // Execute exchange orders
+        uint makerTokenUsed = executeExchangeOrders(_orderData);
+        require(makerTokenUsed <= requiredMakerToken);
+
+        // Check that maker's component tokens in Vault have been incremented correctly
+        for (uint16 i = 0; i < components.length; i++) {
+            uint currentBal = IVault(state.vaultAddress).getOwnerBalance(
```
Comment (asoong) [inline contracts/core/extensions/CoreIssuanceOrder.sol]: Thinking through the possibilities and vulns here - `issue` transfers from the user account if the required component amount is not in the vault.

File: contracts/core/extensions/CoreIssuanceOrder.sol [hunk 2]
```diff
@@ -272,4 +275,57 @@ contract CoreIssuanceOrder is
...
+        // Calculate amount of maker token required
+        // Look into rounding errors
+        uint requiredMakerToken = _fillQuantity.mul(_order.makerTokenAmount).div(_order.quantity);
+
+        // Get Set Token information
+        address[] memory components = ISetToken(_order.setAddress).getComponents();
+        uint[] memory componentUnits = ISetToken(_order.setAddress).getUnits();
+
+        // Calculate amount of component tokens required to issue
+        for (uint16 i = 0; i < components.length; i++) {
```
Comment (asoong) [inline contracts/core/extensions/CoreIssuanceOrder.sol]: I wonder if this is necessary. Issue will fail automatically if the correct component amounts are not either in the vault or the users account

File: contracts/core/extensions/CoreIssuanceOrder.sol [hunk 3]
```diff
@@ -276,4 +292,96 @@ contract CoreIssuanceOrder is
...
+        uint openOrderAmount = _order.quantity.sub(closedOrderAmount);
+        require(
+            openOrderAmount >= _fillQuantity,
+            INVALID_FILL_AMOUNT
+        );
+
+        uint[] memory requiredBalances = new uint[](_order.requiredComponents.length);
+
+        // Calculate amount of maker token required
+        // Look into rounding errors
```
Comment (asoong) [inline contracts/core/extensions/CoreIssuanceOrder.sol]: ?

File: contracts/core/extensions/CoreIssuanceOrder.sol [hunk 4]
```diff
@@ -276,4 +292,96 @@ contract CoreIssuanceOrder is
...
+        uint makerTokenAmountUsed = executeExchangeOrders(_orderData, _order.makerAddress);
+        require(makerTokenAmountUsed <= requiredMakerTokenAmount);
+
+        // Check that maker's component tokens in Vault have been incremented correctly
+        for (i = 0; i < _order.requiredComponents.length; i++) {
+            uint currentBal = IVault(state.vaultAddress).getOwnerBalance(
+                _order.makerAddress,
+                _order.requiredComponents[i]
+            );
+            //require(currentBal >= requiredBalances[i]);
```
Comment (asoong) [inline contracts/core/extensions/CoreIssuanceOrder.sol]: This is pending taker wallet exchange I assume?

File: contracts/core/lib/OrderLibrary.sol [hunk 5]
```diff
@@ -62,16 +68,18 @@ library OrderLibrary {
...
-                _addresses[1], // makerAddress
-                _addresses[2], // makerToken
-                _addresses[3], // relayerAddress
-                _addresses[4], // relayerToken
-                _values[0],    // quantity
-                _values[1],    // makerTokenAmount
-                _values[2],    // expiration
-                _values[3],    // relayerTokenAmount
-                _values[4]     // salt
+                _addresses[0],              // setAddress
```
Comment (asoong) [inline contracts/core/lib/OrderLibrary.sol]: Can we add descriptions to each of these?

File: contracts/core/extensions/CoreIssuanceOrder.sol (hunk 4 above)
Comment (bweick) [reply contracts/core/extensions/CoreIssuanceOrder.sol]: Si

File: contracts/core/extensions/CoreIssuanceOrder.sol (hunk 3 above)
Comment (bweick) [reply contracts/core/extensions/CoreIssuanceOrder.sol]: Added to tech debt

**Pull request merged**